engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java

```diff
@@ -96,125 +86,89 @@ public class StorageManagerTest extends TerasologyTestingEnvironment {
-        context.put(StorageManager.class, esm);
-
-        this.character = entityManager.create();
-        Client client = createClientMock(PLAYER_ID, character);
-        NetworkSystem networkSystem = mock(NetworkSystem.class);
-        when(networkSystem.getMode()).thenReturn(NetworkMode.NONE);
-        when(networkSystem.getPlayers()).thenReturn(List.of(client));
-        context.put(NetworkSystem.class, networkSystem);
-
-        AssetManager assetManager = context.get(AssetManager.class);
-        BlockFamilyDefinitionData data = new BlockFamilyDefinitionData();
-        data.setBlockFamily(SymmetricFamily.class);
-        assetManager.loadAsset(new ResourceUrn("test:testblock"), data, BlockFamilyDefinition.class);
-        assetManager.loadAsset(new ResourceUrn("test:testblock2"), data, BlockFamilyDefinition.class);
-        testBlock = context.get(BlockManager.class).getBlock("test:testblock");
-        testBlock2 = context.get(BlockManager.class).getBlock("test:testblock2");
-
-        context.put(ChunkProvider.class, mock(ChunkProvider.class));
-        WorldProvider worldProvider = mock(WorldProvider.class);
-        when(worldProvider.getWorldInfo()).thenReturn(new WorldInfo());
-        context.put(WorldProvider.class, worldProvider);
-    }
+        character = player.getCharacterEntity();
+        assertThat(character).isNotEqualTo(EntityRef.NULL);
+
+        var clients = ImmutableList.copyOf(network.getPlayers());
+        assertThat(clients).hasSize(1);
+        playerId = clients.get(0).getId();
+
+        testBlock = blockManager.getBlock("test:testblock");
+        testBlock2 = blockManager.getBlock("test:testblock2");
 
-    private Client createClientMock(String clientId, EntityRef charac) {
-        EntityRef clientEntity = createClientEntity(charac);
-        Client client = mock(Client.class);
-        when(client.getEntity()).thenReturn(clientEntity);
-        when(client.getId()).thenReturn(clientId);
-        return client;
+        moduleEnvironment = moduleManager.getEnvironment();
     }
 
-    private EntityRef createClientEntity(EntityRef charac) {
-        ClientComponent clientComponent = new ClientComponent();
-        clientComponent.local = true;
-        clientComponent.character = charac;
-        return entityManager.create(clientComponent);
+    Path getSavePath(Config config) {
+        // TODO: add a more direct way of inspecting StorageManager's path.
+        //   This way of getting the game title (and thus the path) is a bit brittle,
+        //   because world title and game title are not _required_ to be identical.
+        String gameTitle = config.getWorldGeneration().getWorldTitle();
+        return PathManager.getInstance().getSavePath(gameTitle);
     }
 
     @Test
     @Order(1)
-    public void testGetUnstoredPlayerReturnsNewStor() {
-        PlayerStore store = esm.loadPlayerStore(PLAYER_ID);
+    public void testGetUnstoredPlayerReturnsNewStor(StorageManager esm) {
```

Comment:
Why do we have to hand-over the StorageManager for every test case now?

Reply:
I removed some things from being fields on the test class. It was convenient to help get the tests passing one at a time, and it's consistent with how other code would get the storage manager from a Context.